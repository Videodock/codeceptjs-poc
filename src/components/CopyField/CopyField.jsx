import React, { useRef, useState } from 'react';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import * as PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import styles from './CopyField.styles';

const useStyles = makeStyles(styles);

export default function CopyField ({ defaultValue, ...rest }) {
  const { t }                   = useTranslation('common');
  const classes                 = useStyles();
  const [copyText, setCopyText] = useState(t('copy'));
  const inputRef                = useRef();

  const handleCopy = () => {
    inputRef.current.firstElementChild.select();
    document.execCommand("copy");

    setCopyText(t('copied'));
    setTimeout(() => setCopyText(t('copy')), 500);
  };

  return (
    <div className={classes.root}>
      <TextField
        classes={{ root: classes.copyField }}
        size="small"
        defaultValue={defaultValue}
        variant="filled"
        InputProps={{
          ref             : inputRef,
          disableUnderline: true,
          classes         : {
            input           : classes.copyFieldInput,
            adornedEnd      : classes.copyFieldAdornedEnd,
            inputMarginDense: classes.copyFieldInputMarginDense,
          },
          endAdornment: (
            <span onClick={handleCopy} className={classes.copyText}>
              {copyText}
            </span>
          ),
        }}
        {...rest}
      />
    </div>
  );
}

CopyField.propTypes = {
  defaultValue: PropTypes.string,
};
