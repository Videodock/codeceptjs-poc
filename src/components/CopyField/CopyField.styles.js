import { createStyles } from '@material-ui/styles';

export default ({ spacing, palette, shape, breakpoints, typography }) => createStyles({
  root: {
    marginRight: spacing(4),
  },
  copyField: {
    [breakpoints.down('sm')]: {
      width: '100%',
    },
  },
  copyFieldInput: {
    height : 20,
    padding: '8px 12px',
  },
  copyFieldAdornedEnd: {
    color       : palette.primary.main,
    borderRadius: shape.borderRadius,
    fontFamily  : typography.fontFamilyOpenSans,
    fontSize    : typography.pxToRem(14),
  },
  copyFieldInputMarginDense: {
    color                   : palette.text.darkGrey,
    fontWeight              : 'normal',
    [breakpoints.down('sm')]: {
      padding: 8,
    },
  },
  copyText: {
    cursor   : 'pointer',
    textAlign: 'center',
    minWidth : 50,
  },
});
