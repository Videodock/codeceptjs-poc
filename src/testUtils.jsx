import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
