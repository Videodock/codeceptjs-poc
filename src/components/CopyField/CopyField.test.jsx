import React from 'react';
import { render } from '../../testUtils';
import CopyField from './CopyField';

describe('<CopyField />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render((
      <CopyField defaultValue="https://www.trompa.com" />
    ));

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders copy text and default value passed as prop', () => {
    const { getByText, getByDisplayValue } = render((
      <CopyField defaultValue="https://www.trompa.com/" />
    ));

    expect(getByText('Copy')).toBeTruthy();
    expect(getByDisplayValue('https://www.trompa.com/')).toBeTruthy();
  });
});
