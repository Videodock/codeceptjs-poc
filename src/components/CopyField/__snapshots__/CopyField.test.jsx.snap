// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyField /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-copyField-2"
  >
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd makeStyles-copyFieldAdornedEnd-4 MuiInputBase-marginDense MuiFilledInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input makeStyles-copyFieldInput-3 MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense makeStyles-copyFieldInputMarginDense-5"
        type="text"
        value="https://"
      />
      <span
        class="makeStyles-copyText-6"
      >
        Copy
      </span>
    </div>
  </div>
</div>
`;
